src/avitab/apps/MapApp.cpp

```diff
@@ -350,33 +339,48 @@ void MapApp::selectOnlineMaps() {
         return;
     }
 
-    // List the user-defined online maps
-    containerWithClickableList = std::make_unique<ContainerWithClickableCustomList>(
-            &api(), "Select online slippy maps");
-    containerWithClickableList->setListItems(slippyMapNames);
+    std::shared_ptr<img::TileSource> tileSource;
 
-    containerWithClickableList->setSelectCallback([this](int selectedItem) {
-        std::shared_ptr<img::TileSource> newSource;
-        const auto &conf = slippyMaps.at(selectedItem);
+    if (interactive) {
+        // List the user-defined online maps and let the user pick
+        // which map they want to use
+        containerWithClickableList = std::make_unique<ContainerWithClickableCustomList>(
+                &api(), "Select online slippy maps");
+        containerWithClickableList->setListItems(slippyMapNames);
 
-        newSource = std::make_shared<maps::OnlineSlippySource>(
-            conf.servers, conf.url, conf.minZoomLevel, conf.maxZoomLevel,
-            conf.tileWidthPx, conf.tileHeightPx, conf.copyright,
-            conf.protocol);
+        containerWithClickableList->setSelectCallback([this, &tileSource](int selectedItem) {
+            const auto &conf = slippyMaps.at(selectedItem);
 
-        setTileSource(newSource);
-        currentActiveOnlineMap = conf.name;
-    });
+            tileSource = std::make_shared<maps::OnlineSlippySource>(
+                conf.servers, conf.url, conf.minZoomLevel, conf.maxZoomLevel,
+                conf.tileWidthPx, conf.tileHeightPx, conf.copyright,
+                conf.name, conf.protocol);
 
-    containerWithClickableList->setCancelCallback([this] () {
-        api().executeLater([this] () {
-            containerWithClickableList.reset();
-            chooserContainer->setVisible(false);
+            setTileSource(tileSource);
+            currentActiveOnlineMap = conf.name;
         });
-    });
 
-    containerWithClickableList->show(chooserContainer);
-    chooserContainer->setVisible(true);
+        containerWithClickableList->setCancelCallback([this] () {
+            api().executeLater([this] () {
+                containerWithClickableList.reset();
+                chooserContainer->setVisible(false);
+            });
+        });
+
+        containerWithClickableList->show(chooserContainer);
+        chooserContainer->setVisible(true);
+    } else {
+        // If non-interactive selection, pick the first map
+        // found in the mapconfig.json
+        const auto &conf = slippyMaps.at(0);

```

Comment:
At this point in this function we have made enough sanity checks and we know:
1. the mapconfig.json can be parsed correctly
2. there is at least one valid entry in the mapconfig.json. So we pick the first map entry when the program starts (the only time this function is called in non-interactive mode)